### ISSUE TITLE
[BUG] Page.click times out if first match disappears from page

### ISSUE THREAD

**Issue by flaurida** [microsoft/playwright#1691]
**Context:**
- Playwright Version: `0.12.1`
- Operating System: Mac
- Extra: N/A

**Code Snippet**

```javascript
const { chromium } = require("playwright");

let browser;
let page;

beforeAll(async () => {
  browser = await chromium.launch({ headless: false });
  const context = await browser.newContext();
  page = await context.newPage();
});

afterAll(async () => {
  await browser.close();
});

test("disappearingElement", async () => {
  await page.goto("https://ecstatic-booth-c41be5.netlify.com/?duration=300", {
    timeout: 60000,
  });

  await page.click("main > div:first-child button");
  await page.click("label");
  await page.click('text="Close"');
  await page.click("main > div:last-child button");
  await page.click("label");
  await page.click('text="Close"');
});
```

**Describe the bug**

Hi Playwright team! I am experiencing an issue with `page.click`. In particular, it times out in the scenario where 1) a matching element appears, 2) that first matching element disappears from the page, and 3) a new matching element appears. I would expect it to continue looking for the new match and click on that, but rather it just times out even though there is a matching element on the page.

The test above reproduces this issue. Please let me know if I can provide any additional information :)

**Comment by rangeoshun** [comment]
Commenting here to get updates on this :)

Also we used this workaround until fixed:

```js
await page.$eval(selector, (el) => el.click())
```

**Comment by dgozman** [comment]
I think this should be fixed in v0.14. See also issue #1808.
